### PR TITLE
Fix panic on termination

### PR DESCRIPTION
**Pull request by eccles**
[AB#9793](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/9793)

### CODE REVIEW

File: azbus/receiver.go [hunk 1]
```diff
@@ -340,25 +341,32 @@ func (r *Receiver) open() error {
...
 			err := r.receiver.Close(context.Background())
 			if err != nil {
 				azerr := fmt.Errorf("%s: Error closing receiver: %w", r, NewAzbusError(err))
 				r.log.Infof("%s", azerr)
 			}
-			r.receiver = nil
-			for j := range len(r.handlers) {
-				r.handlers[j].Close()
-			}
 			r.handlers = []Handler{}
```
Comment (j-hartley) [inline azbus/receiver.go:368]: Should we cancel and nil out the stashed Context here?

File: azbus/receiver.go [hunk 2]
```diff
@@ -87,6 +87,8 @@ type Receiver struct {
 	receiver *azservicebus.Receiver
 	options  *azservicebus.ReceiverOptions
 	handlers []Handler
+	ctx      context.Context
```
Comment (j-hartley) [inline azbus/receiver.go:90]: Please can you add a comment stating why it is OK to stash a Context in a struct (a wise man once told me that was an anti-pattern)

File: azbus/receiver.go (hunk 1 above)
Comment (eccles) [reply azbus/receiver.go:366]: yes fixed

File: azbus/receiver.go (hunk 2 above)
Comment (eccles) [reply azbus/receiver.go]: removed context from struct

**Review by j-hartley — APPROVED** [review]

**Pull request merged**